GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Submodules.cs

```diff
@@ -342,7 +350,7 @@ private void CreateSubmoduleNodes(SubmoduleInfoResult result, GitModule threadMo
                 string GetSubmoduleSuperPath(string submodulePath)
                 {
                     var superPath = modulePaths.Find(path => submodulePath != path && submodulePath.Contains(path));
-                    Trace.Assert(superPath is not null);
+                    Assumes.NotNull(superPath);
```

Comment:
Various 'soft' assertions (i.e. showing debug assertion messages, only in debug builds) have been changed into hard throws in release builds.